tools/asset-inventory/asset_inventory/import_pipeline.py

```diff
@@ -0,0 +1,394 @@
+        if write_disposition == 'WRITE_APPEND':
+            job_config.schema_update_options = [
+                bigquery.job.SchemaUpdateOption.ALLOW_FIELD_ADDITION]
+        # use load_time as a timestamp.
+        job_config.time_partitioning = bigquery.table.TimePartitioning(
+            field='timestamp')
+        job_config.schema = schemas[key_name]
+        job_config.source_format = bigquery.SourceFormat.NEWLINE_DELIMITED_JSON
+        try:
+            load_job = self.bigquery_client.load_table_from_uri(
+                object_paths,
+                table_ref,
+                location=self.dataset_location,
+                job_config=job_config)
+            self.load_jobs[key_name] = load_job
+        except BadRequest as e:
+            logging.error('error in load_job %s, %s, %s, %s',
+                          str(object_paths), str(table_ref),
+                          str(self.dataset_location),
+                          str(job_config.to_api_repr()))
+            raise e
+
+    def finish_bundle(self):
+        self.bigquery_client = None
+        # wait for the load jobs to complete
+        for _, load_job in self.load_jobs.items():
+            try:
+                load_job.result()
+            except BadRequest as e:
+                logging.error('error in load_job %s', load_job.self_link)
+                raise e
+
+
+class ImportAssetOptions(PipelineOptions):
+    """Required options.
+
+    All options are required, but are not marked as such to support creation
+    of Dataflow templates.
+    """
+
+    @classmethod
+    def _add_argparse_args(cls, parser):
+        parser.add_value_provider_argument(
+            '--group_by',
+            default=StaticValueProvider(str, 'ASSET_TYPE'),
+            choices=['ASSET_TYPE', 'ASSET_TYPE_VERSION'],
+            help='How to group exported resources into Bigquery tables.')
+
+        parser.add_value_provider_argument(
+            '--write_disposition',
+            default=StaticValueProvider(str, 'WRITE_APPEND'),
+            choices=['WRITE_APPEND', 'WRITE_EMPTY'],
+            help='To append to or overwrite BigQuery tables..')
+
+        parser.add_value_provider_argument(
+            '--input', help='A glob of all input asset json files to process.')
+
+        parser.add_value_provider_argument(
+            '--stage',
+            help='GCS location to write intermediary BigQuery load files.')
+
+        parser.add_value_provider_argument(
+            '--load_time',
+            default=StaticValueProvider(str, datetime.now().isoformat()),
+            help='Load time of the data (YYYY-MM-DD[HH:MM:SS])).')
+
+        parser.add_value_provider_argument(
+            '--dataset', help='BigQuery dataset to load to.')
+
+
+def run(argv=None):
+    """Construct the pipeline."""
+
+    options = ImportAssetOptions(argv)
+
+    p = beam.Pipeline(options=options)
+
+    # Delete bigquery dataset on pipeline start (not sure about that).
+    deleted_tables = (
```

Comment:
Consider refactoring `DeleteDataSetTables` to be a `PTransform` that accepts a `PBegin` rather than creating this hack-y meaningless PCollection with `beam.Create([True])` that is just discarded to trigger performing an action at the beginning of the pipeline.

Reply:
I don't see how that can work.   PBegin seems only usable for Create or Read transforms and I'm a Create transform. 

To somehow use a read transform I have to create a new custom source that extends BoundedSource which implements the methods estimate_size,get_range_tracker,read, and split.  This Source wouldn't produce any output other then "True" when the tables are deleted.  That seems like a screen or more of code just to have a processes run at the start of the pipline right?   I think what I have is cleaner.

Reply:
Ok point taken. Can you change the `beam.Create(
[True])` to `beam.Create([None])` as the initial `True` has no meaning and is just discarded? Also add a quick comment that this is a dummy PCollection to trigger this table clean up to avoid confusion.

Reply:
done